api/tests/routers/test_ops.py

```diff
@@ -0,0 +1,93 @@
+import os
+from fastapi.testclient import TestClient
+from unittest.mock import patch
+import main
+
+client = TestClient(main.app)
+
+
+# Test that the version endpoint returns unknown if the GIT_SHA environment variable is not set
+def test_version_with_no_GIT_SHA():
+    response = client.get("/version")
+    assert response.status_code == 200
+    assert response.json() == {"version": "unknown"}
+
+
+# Test that the version endpoint returns the GIT_SHA environment variable if it is set
+@patch.dict(os.environ, {"GIT_SHA": "foo"}, clear=True)
+def test_version_with_GIT_SHA():
+    response = client.get("/version")
+    assert response.status_code == 200
+    assert response.json() == {"version": "foo"}
+
+
+# Test that the healthcheck endpoint returns an error when the dynamoDB host is empty
+@patch.dict(os.environ, {"DYNAMODB_HOST": ""}, clear=True)
+def test_healthcheck_failure_host_empty():
+    response = client.get("/healthcheck")
+    assert response.status_code == 200
+    expected_val = {
+        "status": "ERROR",
+        "message": "Not able to connect to the database",
+    }
+    assert response.json() == expected_val
+
+
+# Test that the healthcheck endpoint returns an error when the dynamoDB host provided is wrong
+@patch.dict(os.environ, {"DYNAMODB_HOST": "foo_db"}, clear=True)
+def test_healthcheck_failure_wrong_host():
+    response = client.get("/healthcheck")
+    assert response.status_code == 200
+    expected_val = {
+        "status": "ERROR",
+        "message": "Not able to connect to the database",
+    }
+    assert response.json() == expected_val
+
+
+# Test that the healthcheck endpoint returns an error when the table name provided is empty
+@patch.dict(os.environ, {"TABLE_NAME": ""}, clear=True)
+def test_healthcheck_failure_empty_table_name():
+    response = client.get("/healthcheck")
+    assert response.status_code == 200
+    # expected_val = {
+    #     "status": "ERROR",
+    #     "message": "Parameter validation failed:\nInvalid length for parameter TableName, value: 0, valid min length: 3",
+    # }
```

Comment:
Just a minor nit - could you remove this before merging.

Reply:
Removed! Totally did not see it - I guess you stare at it for so long that it becomes invisible. Thanks for noticing it.

Reply:
:100: